script.js

```diff
@@ -0,0 +1,140 @@
+const headerNavbar = document.getElementById('navbarSupportedContent');
+const resultBlock = document.getElementById('results');
+const pages = document.querySelector('.pagination');
+const search = document.querySelector('.form-control');
+const loader = document.querySelector('.loader');
+let currentPage = 1;
+const selectedCategory = localStorage.getItem('category') || KEYS_VARIABLES.categories.people;
+
+
+function getData(path) {
+    resultBlock.innerHTML = null;
+    loader.classList.remove(KEYS_VARIABLES.modificators.loader_hide);
+    return fetch(`https://swapi.dev/api/${path}`).then((res) => res.json());
+}
+
+function getSelectedCategory() {
+    let category = null;
+    headerNavbar.querySelectorAll('input').forEach(item => {
+        if (item.checked) {
+            category = item.getAttribute('id');
+        }
+    })
+    return category;
+}
+
+pages.addEventListener('click', ({target}) => {
+    const pageItem = target.closest('.page-item');
+    if (!pageItem.classList.contains('.active')) {
+        pages.querySelectorAll('li').forEach(item => item.classList.remove('active'));
+        pageItem.classList.add('active');
+        const selectedCategory = getSelectedCategory();
+        getData(`${selectedCategory}/?page=${pageItem.getAttribute('data-index')}`)
+            .then((data) => displayResults(data, selectedCategory));
+    }
+})
+
+resultBlock.addEventListener('click', (event) => {
+    if (event.target.closest('.btn')) {
+        $(document.querySelectorAll('.collapse')).collapse('hide');
```

Comment:
remove jQuery

Reply:
Without jquery I found only one solution to close collapse without missing transition
 ```suggestion
 function collapseListener(event){
    if (event.target.closest('.btn')) {
        resultBlock.removeEventListener('click',collapseListener)
        resultBlock.querySelectorAll('.btn').forEach(item=>{
            if(!item.classList.contains('collapsed')){
                item.click();
            }
        })
        resultBlock.addEventListener('click',collapseListener)
    }
}
resultBlock.addEventListener('click', collapseListener);
```